modules/gdscript/gdscript_analyzer.cpp

```diff
@@ -4350,15 +4940,357 @@ void GDScriptAnalyzer::reduce_unary_op(GDScriptParser::UnaryOpNode *p_unary_op)
 	} else {
 		bool valid = false;
 		result = get_operation_type(p_unary_op->variant_op, operand_type, valid, p_unary_op);
-
 		if (!valid) {
 			push_error(vformat(R"(Invalid operand of type "%s" for unary operator "%s".)", operand_type.to_string(), Variant::get_operator_name(p_unary_op->variant_op)), p_unary_op);
 		}
+		unary_op_nullity_guard(p_unary_op);
 	}
 
 	p_unary_op->set_datatype(result);
 }
 
+bool GDScriptAnalyzer::unary_op_nullity_guard(GDScriptParser::UnaryOpNode *p_unary_op) {
+	if (!p_unary_op->operand->get_datatype().is_nullable || p_unary_op->operation == GDScriptParser::UnaryOpNode::OP_LOGIC_NOT) {
+		// Nullable verification only makes sense for unary operators if the operator is not the logic negation (!foo)
+		return true;
+	}
+
+	GDScriptParser::IdentifierNode *identifier = static_cast<GDScriptParser::IdentifierNode *>(p_unary_op->operand);
+	push_nullable_error(
+			R"*(Potentially unsafe for nullable "%s" to be used with the "%s" unary operator.)*",
+			{ p_unary_op->operand },
+			identifier,
+			p_unary_op->operand->type == GDScriptParser::Node::IDENTIFIER ? String(identifier->name) : p_unary_op->operand->get_datatype().to_string(),
+			Variant::get_operator_name(p_unary_op->variant_op));
+	return false;
+}
+
+Vector<GDScriptParser::IdentifierNode *> GDScriptAnalyzer::deduce_nullable_narrowing(GDScriptParser::ExpressionNode *p_node, bool p_should_be_false) {
```

Comment:
What does `p_should_be_false` mean? I took a stab in the dark and renamed it to `p_current_op_is_wide`, which is a double-negative-ish of `p_current_op_is_not_narrowed`. Is this the correct understanding?

Should it be named and simplified? i.e. `p_should_be_false` -> `p_current_op_is_narrowed`?